### PR TITLE
Rewrite NumerAPI

### PR DESCRIPTION
**Pull request by Californian**
Rewrote to work with new endpoints, clarify, and secure code.

### CODE REVIEW

**Comment by xanderdunn** [comment]
@Californian Don't forget to assign a reviewer

**Comment by xanderdunn** [comment]
Oh, I see.  You weren't able to add a reviewer because of permissions?

**Comment by Californian** [comment]
By the way, I wasn't sure if it would make sense to remove `numpy` as a dependency (and use normal `list`s instead of `numpy.ndarray`s for everything). If that would make more sense, let me know and I'll change it accordingly.

File: README.md [hunk 1]
```diff
@@ -1,76 +1,199 @@
-# Numerai Python API
-Automatically download and upload data for the Numerai machine learning competition
-
-This library is a client to the Numerai API. The interface is programmed in Python and allows downloading the training data, uploading predictions, and accessing some user information. Some parts of the code were taken from [numerflow](https://github.com/ChristianSch/numerflow) by ChristianSch. Visit his [wiki](https://github.com/ChristianSch/numerflow/wiki/API-Reverse-Engineering), if you need further information on the reverse engineering process.
+# NumerAI Python API
```
Comment (xanderdunn) [inline README.md]: NumerAI -> Numerai.  It's never written "NumerAI".

File: README.md [hunk 2]
```diff
@@ -1,76 +1,199 @@
...
+[wiki](https://github.com/ChristianSch/numerflow/wiki/API-Reverse-Engineering), 
+if you need further information on the reverse engineering process.
 
 If you encounter a problem or have suggestions, feel free to open an issue.
 
 # Installation
-This library supports both Python 2 and 3.  Clone this repository, then `cd` into this repository's directory.  Then `pip install -e .`
+This library supports both Python 2 and 3.  Clone this repository, then `cd` 
+into this repository's directory.  Then `pip install -e .` Using a Virtual 
+Environment is always recommended.
```
Comment (xanderdunn) [inline README.md]: I'd remove this mention of a virtual environment.  It's outside the scope of this README.

**Comment by xanderdunn** [comment]
@Californian Overall these changes are very good.  Thanks for doing this.  It looks much better now.

**Comment by xanderdunn** [comment]
@Californian I'd leave it as numpy for now.  Numpy is what all our users will be using to participate in our tournament, so they're familiar with it.

**Comment by Californian** [comment]
@xanderdunn that's what I figured (re: numpy).

**Pull request merged**